server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -128,7 +128,7 @@ public void init() throws IOException {
      * @return Map of segment filename to uploaded filename with checksum
      * @throws IOException if there were any failures in reading the metadata file
      */
-    private Map<String, UploadedSegmentMetadata> readLatestMetadataFile() throws IOException {
+    public Map<String, UploadedSegmentMetadata> readLatestMetadataFile() throws IOException {
```

Comment:
Metadata file should not be known outside of `RemoteSegmentStoreDirectory`. Any reason of making it public?

Reply:
Got it. Currently we're reading the metadata file to compute the segments diff in `SegmentReplicationTarget`. 
With https://github.com/opensearch-project/OpenSearch/issues/7802, we should be able to clear up the dependency and move it back to private.